aiocache/decorators.py

```diff
@@ -55,7 +55,9 @@ def cached_decorator(func):
 
             try:
                 if await cache_instance.exists(cache_key):
-                    return await cache_instance.get(cache_key)
+                    result = await cache_instance.get(cache_key)
+                    await cache_instance.close()
```

Comment:
I see where you are coming from, but I don't think this "delay" is as big of a deal as you are making it out to be.  Since the code is using asyncio it is not as if the program will be idle whilst closing the connection, the event loop will context switch to another task.

I have made the changes and the error messages have now returned.  I would advise either reverting back to how I originally wrote it (awaiting the close), or reusing the same cache instance within a decorator.  Ultimately it is up to you.